contrib/valgrind.supp

```diff
@@ -41,3 +45,34 @@
    ...
    fun:_ZN7leveldbL14InitDefaultEnvEv
 }
+{
+   Suppress boost::filesystem warning
```

Comment:
Is there a fix or bug report for this?

Reply:
Not that I've been able to find. I created the suppressions on a plain Ubuntu 18.04.2 LTS machine. As non-exotic as it gets :)

Reply:
Full output:

```
==29772== Invalid read of size 16                                       
==29772==    at 0x7712ABD: __wcsnlen_sse4_1 (strlen.S:194)       
==29772==    by 0x7700458: wcsnrtombs (wcsnrtombs.c:58)           
==29772==    by 0x6DB2E6D: std::codecvt<wchar_t, char, __mbstate_t>::do_out(__mbstate_t&, wchar_t const*, wchar_t const*, wchar_t const*&, char*, char*, char*&) const (in /usr/lib/x86_64-linux-g
nu/libstdc++.so.6.0.25)
==29772==    by 0x50551B1: boost::filesystem::path_traits::convert(wchar_t const*, wchar_t const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::codecvt<
wchar_t, char, __mbstate_t> const&) (in /usr/lib/x86_64-linux-gnu/libboost_filesystem.so.1.65.1)
==29772==    by 0x505563A: boost::filesystem::detail::unique_path(boost::filesystem::path const&, boost::system::error_code*) (in /usr/lib/x86_64-linux-gnu/libboost_filesystem.so.1.65.1)
==29772==    by 0x904E75: unique_path (operations.hpp:723)
==29772==    by 0x904E75: DirIsWritable(boost::filesystem::path const&) (system.cpp:122)        
==29772==    by 0x4CE8A0: util_tests::test_DirIsWritable::test_method() (util_tests.cpp:1613)
==29772==    by 0x4CEF98: util_tests::test_DirIsWritable_invoker() (util_tests.cpp:1601)
==29772==    by 0x58EF2CD: boost::detail::function::function_obj_invoker0<boost::detail::forward, int>::invoke(boost::detail::function::function_buffer&) (in /usr/lib/x86_64-linux-gnu/libboost_u
nit_test_framework.so.1.65.1)                    
==29772==    by 0x58EE77C: boost::execution_monitor::catch_signals(boost::function<int ()> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.65.1)
==29772==    by 0x58EE860: boost::execution_monitor::execute(boost::function<int ()> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.65.1)
==29772==    by 0x58EEFDC: boost::execution_monitor::vexecute(boost::function<void ()> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.65.1)
==29772==  Address 0x12b82050 is 0 bytes after a block of size 80 alloc'd
==29772==    at 0x4C3017F: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==29772==    by 0x6E3BCAA: std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >::_M_mutate(unsigned long, unsigned long, wchar_t const*, unsigned long) (in /u
sr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)
==29772==    by 0x6E3CB02: std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >::_M_replace(unsigned long, unsigned long, wchar_t const*, unsigned long) (in /
usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)                                                                                                                                                     ==29772==    by 0x5054F11: boost::filesystem::path_traits::convert(char const*, char const*, std::__cxx11::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >&, std::codec
vt<wchar_t, char, __mbstate_t> const&) (in /usr/lib/x86_64-linux-gnu/libboost_filesystem.so.1.65.1)
==29772==    by 0x50556F7: boost::filesystem::detail::unique_path(boost::filesystem::path const&, boost::system::error_code*) (in /usr/lib/x86_64-linux-gnu/libboost_filesystem.so.1.65.1)
==29772==    by 0x904E75: unique_path (operations.hpp:723)
==29772==    by 0x904E75: DirIsWritable(boost::filesystem::path const&) (system.cpp:122)
==29772==    by 0x4CE8A0: util_tests::test_DirIsWritable::test_method() (util_tests.cpp:1613)
==29772==    by 0x4CEF98: util_tests::test_DirIsWritable_invoker() (util_tests.cpp:1601)
==29772==    by 0x58EF2CD: boost::detail::function::function_obj_invoker0<boost::detail::forward, int>::invoke(boost::detail::function::function_buffer&) (in /usr/lib/x86_64-linux-gnu/libboost_u
nit_test_framework.so.1.65.1)
==29772==    by 0x58EE77C: boost::execution_monitor::catch_signals(boost::function<int ()> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.65.1)
==29772==    by 0x58EE860: boost::execution_monitor::execute(boost::function<int ()> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.65.1)
==29772==    by 0x58EEFDC: boost::execution_monitor::vexecute(boost::function<void ()> const&) (in /usr/lib/x86_64-linux-gnu/libboost_unit_test_framework.so.1.65.1)
==29772==
{                                                                                     
   <insert_a_suppression_name_here>
   Memcheck:Addr16                                          
   fun:__wcsnlen_sse4_1         
   fun:wcsnrtombs
   fun:_ZNKSt7codecvtIwc11__mbstate_tE6do_outERS0_PKwS4_RS4_PcS6_RS6_
   fun:_ZN5boost10filesystem11path_traits7convertEPKwS3_RNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERKSt7codecvtIwc11__mbstate_tE
   fun:_ZN5boost10filesystem6detail11unique_pathERKNS0_4pathEPNS_6system10error_codeE
   fun:unique_path
   fun:_Z13DirIsWritableRKN5boost10filesystem4pathE
   fun:_ZN10util_tests18test_DirIsWritable11test_methodEv
   fun:_ZN10util_testsL26test_DirIsWritable_invokerEv
   fun:_ZN5boost6detail8function21function_obj_invoker0INS0_7forwardEiE6invokeERNS1_15function_bufferE
   fun:_ZN5boost17execution_monitor13catch_signalsERKNS_8functionIFivEEE
   fun:_ZN5boost17execution_monitor7executeERKNS_8functionIFivEEE
   fun:_ZN5boost17execution_monitor8vexecuteERKNS_8functionIFvvEEE
}
```

Reply:
Added reference to related issue.